libs/hpc-data/src/lib/flows.ts

```diff
@@ -0,0 +1,503 @@
+  sourceCountries: t.array(t.string),
+  sourceOrganizations: t.array(t.string),
+  sourceUsageYears: t.array(t.string),
+  sourceProjects: t.array(t.string),
+  sourcePlans: t.array(t.string),
+  sourceGlobalClusters: t.array(t.string),
+  sourceEmergencies: t.array(t.string),
+  includeChildrenOfParkedFlows: t.boolean,
+});
+
+const FLOW_FILTERS = t.partial({
+  flowFilters: t.partial({
+    id: t.array(t.number),
+    activeStatus: t.boolean,
+    status: t.string,
+    amountUSD: t.number,
+    type: t.string,
+    restricted: t.boolean,
+  }),
+  flowObjectFilters: t.array(
+    t.type({ objectID: t.number, direction: t.string, objectType: t.string })
+  ),
+  commitment: t.boolean,
+  carryover: t.boolean,
+  paid: t.boolean,
+  pledged: t.boolean,
+  parked: t.boolean,
+  pass_through: t.boolean,
+  standard: t.boolean,
+  flowCategoryFilters: t.array(t.type({ id: t.number, group: t.string })),
+  pending: t.boolean,
+  includeChildrenOfParkedFlows: t.boolean,
+  nestedFlowFilters: t.partial({
+    reporterRefCode: t.string,
+    legacyID: t.number,
+    sourceSystemID: t.string,
+  }),
+});
+// TODO: transform to arrays of
+export type FlowFilters = t.TypeOf<typeof FLOW_FILTERS>;
+const AbortSignalType = new t.Type<AbortSignal, AbortSignal, unknown>(
+  'AbortSignal',
+  (input: unknown): input is AbortSignal => input instanceof AbortSignal,
+  (input, context) =>
+    input instanceof AbortSignal ? t.success(input) : t.failure(input, context),
+  t.identity
+);
+
+export const NESTED_FLOW_FILTERS = t.partial({
+  reporterRefCode: t.string,
+  legacyID: t.number,
+  sourceSystemID: t.string,
+});
+export type NestedFlowFilters = t.TypeOf<typeof NESTED_FLOW_FILTERS>;
+
+export type FlowFiltersREST = t.TypeOf<typeof FILTERS>;
+export const SEARCH_FLOWS_PARAMS = t.partial({
+  limit: t.number,
+  prevPageCursor: t.number,
+  nextPageCursor: t.number,
+  sortOrder: t.string,
+  sortField: t.string,
+  ...FLOW_FILTERS.props,
+  signal: AbortSignalType,
+});
+export type SearchFlowsParams = t.TypeOf<typeof SEARCH_FLOWS_PARAMS>;
+
+export const BULK_REJECT_PENDING_FLOWS_PARAMS = t.type({
+  flows: t.array(
+    t.type({
+      id: t.number,
+      versionID: t.number,
+    })
+  ),
+});
+export type BulkRejectPendingFlowsParams = t.TypeOf<
+  typeof BULK_REJECT_PENDING_FLOWS_PARAMS
+>;
+
+export const BULK_REJECT_PENDING_FLOWS_RESULT = t.array(
+  t.type({
+    id: t.number,
+    versionID: t.number,
+  })
+);
+
+export type BulkRejectPendingFlowsResults = t.TypeOf<
+  typeof BULK_REJECT_PENDING_FLOWS_RESULT
+>;
+
+export const GET_TOTAL_AMOUNT_USD_PARAMS = t.partial({
+  ...FLOW_FILTERS.props,
+  signal: AbortSignalType,
+});
+export type GetTotalAmountUSDParams = t.TypeOf<
+  typeof GET_TOTAL_AMOUNT_USD_PARAMS
+>;
+
+export const GET_TOTAL_AMOUNT_USD_RESULT = t.type({
+  searchFlowsTotalAmountUSD: t.type({
+    totalAmountUSD: t.string,
+  }),
+});
+export type GetTotalAmountUSDResult = t.TypeOf<
+  typeof GET_TOTAL_AMOUNT_USD_RESULT
+>;
```

Comment:
Well, TotalAmountUSD is giving problems because it consumes a lot, and since is not feature parity, we will take it out for the moment. So this will answer this comment